### ISSUE TITLE
Chart Annotations

### ISSUE THREAD

**Issue by pavelgruba** [DevExpress/DevExtreme#7258]
# The Problem 

In certain cases, visualized data needs to be annotated with an image or text block. The Chart already provides a similar functionality in the form of labels and tooltips, but they are anchored to series points.

# The Proposed Solution

We plan to implement image and text annotations that you can anchor to any chart element or leave unattached.

Chart annotations will have the following API:
```js
let chartOptions = {
  annotations: [{ 
    // text annotation 
    // particular combination of options depend on your UI design
  }, {
    // image annotation 
    // particular combination of options depend on your UI design
  }],
  annotationStyle: { 
    label: {
        // appearance settings that apply to all text annotations
    }, 
    image: {
        // appearance settings that apply to all image annotations
    } 
  } 
} 
```
This API gives us a capability to add more annotation types in the future.   

Each annotation can have the following options. Which of the coordinate fields to use depends on the use-case.

```js
{ 
  visible: true,
  x: 0, // pixel coordinate 
  y: 0, // pixel coordinate 
  value: 0, // value coordinate 
  argument: 0, // argument coordinate
  axis: "value_axis_name", // the value axis by which to track the value coordinate
                           // (for a chart with multiple value axes) 
  series: "series_name" // the series by which to track the argument coordinate
                        // (when multiple series points correspond to a single argument)
} 
```

## Add an unanchored annotation 

```js
annotations: [{
  x: 100,
  y: 200
}]
```

## Anchor an annotation to a chart coordinate 

```js
annotations: [{ 
  value: 15,  
  argument: new Date(2018, 1, 16),
  axis: "value_axis_2" // in a chart with multiple value axes
}]
```

## Display an annotation on the axis 

```js
annotations: [{ 
  // this annotation will be shown on the value axis 
  value: 15,  
  axis: "value_axis_2" // in a chart with multiple value axes
}, { 
  // this annotation will be shown on the argument axis 
  argument: new Date(2018, 1, 16)
}]
```

## Anchor an annotation to a series 

```js
annotations: [{ 
  value: 15, 
  argument: "Apple",
  axis: "value_axis_2", // in a chart with multiple value axes
  series: "series_name"
}]
```

## Mixed anchoring 

You can use pixel and chart coordinates simultaneously:

```js
annotations: [{ 
  y: 150,  
  argument: new Date(2018, 1, 16)
}]
```

# We Need Your Feedback

## Take a Quick Poll

[What other annotation types would be useful for your chart?](https://docs.google.com/forms/d/e/1FAIpQLSe7rg1xkR6pxGRME7pdoQO_B2Ulu_8t0t53F-ha7DaVPUye5A/viewform?usp=sf_link)

# Get Notified of Updates

Subscribe to this thread or to our [Facebook](https://www.facebook.com/DevExpress.DevExtreme/) and [Twitter](https://twitter.com/devextreme) accounts for updates on this topic.

**Comment by GlenHerman** [comment]
Ability for the user to right mouse click, add annotation, change text and anchor with drag to any location on the chart

**Comment by pavelgruba** [comment]
We will consider implementing such a functionality in future releases.

**Comment by pavelgruba** [comment]
The Chart Annotations feature is now available in the [v19.1](https://github.com/DevExpress/DevExtreme/releases/tag/19.1.3) release. Based on the feedback, we've changed the proposed API. Please refer to [this documentation topic](https://js.devexpress.com/Documentation/ApiReference/Data_Visualization_Widgets/dxChart/Configuration/annotations/) and [the demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Charts/Annotation/jQuery/Light/) for more information.
If you have questions or find any bugs, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
